[163] created Delete-Event

RockTF opened the pull request:
"Merge is required after task https://github.com/OlhenShu/ProjectStage_Team1_May/issues/157." 
PR - https://github.com/GreenCityProject/GreenCityMVP21_team1_May/pull/64

BerserkrMM committed:
[157] create event functionality & infrastructure (draft)

BerserkrMM committed:
[157] create event functionality works (draft)

BerserkrMM committed:
[157] liquibase changeset added (draft)

BerserkrMM committed:
[157] small fix & mapper added (draft)

BerserkrMM committed:
[157] added notification sending & string validation (exclusion html tags)

BerserkrMM committed:
[157] fixed event_id for additionalImages. fixed nulls for organizer

BerserkrMM committed:
[157] small fix

BerserkrMM committed:
[157] automated title image selection. set default image. Common ValidationException added

BerserkrMM committed:
[157] added validations. handled validation failure

BerserkrMM committed:
[157] testing started

df7734 committed:
[161]-viable_update_for_event

df7734 committed:
[161]-merge-with-157-2830b5b

df7734 committed:
[161]-viable_update_after_merge

BerserkrMM committed:
[157] add test case for EventController (201Created handling)

IliaWithHat committed:
[333] rewriting

IliaWithHat committed:
[333] add method that sends notification when comment liked

IliaWithHat committed:
[330] add method that sends notification when new comment was added

RockTF committed:
163-Delete-Event - Add delete functionality to Event service

This commit includes the implementation of a delete function in the Event service. Corresponding updates are made to ErrorMessage for handling event not found errors. New files like EventService, EventServiceImpl, EventRepo, Event entity and EventController are added, with logic for deleting an event based on eventId and user email.

RockTF committed:
163-Delete-Event - Add delete functionality to Event service

This commit includes the implementation of a delete function in the Event service. Corresponding updates are made to ErrorMessage for handling event not found errors. New files like EventService, EventServiceImpl, EventRepo, Event entity and EventController are added, with logic for deleting an event based on eventId and user email.

RockTF committed:
163-Delete-Event - Add delete functionality to Event service

This commit includes the implementation of a delete function in the Event service. Corresponding updates are made to ErrorMessage for handling event not found errors. New files like EventService, EventServiceImpl, EventRepo, Event entity and EventController are added, with logic for deleting an event based on eventId and user email.

RockTF committed:
163-Delete-Event - Add EventServiceImplTest for delete functionality

This commit introduces a new test file, EventServiceImplTest, to test the delete functionality in the Event service. Several scenarios are covered including the event not found, user not having permission to delete, and successful deletion. These tests ensure that the service's delete function is working as expected.

RockTF committed:
163-Delete-Event - Add EventServiceImplTest for delete functionality

This commit introduces a new test file, EventServiceImplTest, to test the delete functionality in the Event service. Several scenarios are covered including the event not found, user not having permission to delete, and successful deletion. These tests ensure that the service's delete function is working as expected.

RockTF committed:
163-Delete-Event - Add EventServiceImplTest for delete functionality

This commit introduces a new test file, EventServiceImplTest, to test the delete functionality in the Event service. Several scenarios are covered including the event not found, user not having permission to delete, and successful deletion. These tests ensure that the service's delete function is working as expected.

df7734 committed:
[161] minor changes

df7734 committed:
Merge branch 'feature/157-create_event' of https://github.com/GreenCityProject/GreenCityMVP21_team1_May into feature/161-edit_event

df7734 committed:
[161] fix organizer has no permission

RockTF commented:
/hold

BerserkrMM committed:
[157] added event already exist check;
added validations;
reorganized validations (entity&dto);
small fixes.

BerserkrMM committed:
Merge branch 'dev_java21' into feature/157-create_event

# Conflicts:
#	dao/src/main/resources/db/changelog/db.changelog-master.xml

BerserkrMM committed:
Merge branch 'dev_java21' into feature/157-create_event

BerserkrMM committed:
Merge remote-tracking branch 'origin/feature/157-create_event' into feature/157-create_event

df7734 committed:
[161] merge with 157 (PR)

df7734 committed:
[161] minor changes

RockTF committed:
163-Delete-Event - Add delete event functionality test in EventController

This commit inserts a new test "deleteEvent_validRequest_200Ok" in EventControllerTest.java which verifies the delete event functionality. The update also includes import statements for the delete method and eq matcher. It ensures that the delete feature is working correctly, returning a status of OK for a valid delete event request.

RockTF committed:
163-Delete-Event - Add delete event functionality test in EventController

This commit inserts a new test "deleteEvent_validRequest_200Ok" in EventControllerTest.java which verifies the delete event functionality. The update also includes import statements for the delete method and eq matcher. It ensures that the delete feature is working correctly, returning a status of OK for a valid delete event request.

RockTF committed:
Merge remote-tracking branch 'origin/163-Delete-Event' into 163-Delete-Event

# Conflicts:
#	core/src/main/java/greencity/controller/EventController.java
#	dao/src/main/java/greencity/entity/Event.java
#	dao/src/main/java/greencity/repository/EventRepo.java
#	service-api/src/main/java/greencity/service/EventService.java
#	service/src/main/java/greencity/service/EventServiceImpl.java
#	service/src/test/java/greencity/service/EventServiceImplTest.java

RockTF committed:
Reposition event deletion method in EventServiceImpl

The deletion method for events has been moved upwards within the EventServiceImpl file. The logic and functionality remain unchanged. This rearrangement improves the readability and organization of the code.

RockTF commented:
/hold cancel

BerserkrMM committed:
[157] added custom AlreadyExistException;
fixed event.equals bug;
fixed BigDecimal precision bug;
small fixes.

IliaWithHat committed:
Merge pull request #64 from GreenCityProject/feature/157-create_event

[157] Create Event

df7734 committed:
Merge branch 'dev_java21' into feature/161-edit_event

df7734 committed:
Merge pull request #65 from GreenCityProject/feature/161-edit_event

[161] Edit event

BerserkrMM commented on service/src/main/java/greencity/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,216 @@
+     */
+    @Override
+    public void delete(Long eventId, String email) {
+        UserVO userVO = restClient.findByEmail(email);
```
You can use userService.findByEmail() instead sending http request to User Service

IliaWithHat committed:
Merge branch 'refs/heads/dev_java21' into feature/333-Comment_was_liked_notification

# Conflicts:
#	core/src/main/java/greencity/config/SecurityConfig.java

IliaWithHat committed:
Merge pull request #58 from GreenCityProject/feature/333-Comment_was_liked_notification

[333] comment was liked notification

IliaWithHat committed:
Merge branch 'refs/heads/dev_java21' into feature/330-News_article_was_commented_notification

# Conflicts:
#	core/src/main/java/greencity/config/SecurityConfig.java

IliaWithHat committed:
Merge pull request #60 from GreenCityProject/feature/330-News_article_was_commented_notification

[330] News article was commented notification

BerserkrMM commented:
add `"/event/delete/{eventId}",` to security config

RockTF committed:
Add delete event to SecurityConfig

An option for deleting an event was added to the SecurityConfig file. This new line of code allows users who have any of the USER, ADMIN, MODERATOR or UBS_EMPLOYEE roles to delete an event using the '/events/delete/{eventId}' HTTP request.

RockTF committed:
Add delete event to SecurityConfig

An option for deleting an event was added to the SecurityConfig file. This new line of code allows users who have any of the USER, ADMIN, MODERATOR or UBS_EMPLOYEE roles to delete an event using the '/events/delete/{eventId}' HTTP request.

RockTF committed:
Replace restClient with userService in EventServiceImpl

The restClient used in the EventServiceImpl for finding users by email has been replaced with userService. The method findByEmail from userService was used instead of restClient in the delete method in EventServiceImpl and in its corresponding tests, ensuring tests reflect actual application behavior.

RockTF committed:
Replace restClient with userService in EventServiceImpl

The restClient used in the EventServiceImpl for finding users by email has been replaced with userService. The method findByEmail from userService was used instead of restClient in the delete method in EventServiceImpl and in its corresponding tests, ensuring tests reflect actual application behavior.

RockTF committed:
Remove author tag from EventController delete method documentation

The author tag from the delete method documentation in the EventController class has been removed. The author information was irrelevant and unneeded in this file.

RockTF committed:
Remove author tag from EventController delete method documentation

The author tag from the delete method documentation in the EventController class has been removed. The author information was irrelevant and unneeded in this file.

RockTF committed:
Update delete event method documentation

The documentation for the "delete event" method in the EventController class has been updated. The new doc now includes further details about the method parameters and expected return value. It explains that the method deletes an event based on a given Event ID and returns server's response indicating the operation status.

RockTF committed:
Update delete event method documentation

The documentation for the "delete event" method in the EventController class has been updated. The new doc now includes further details about the method parameters and expected return value. It explains that the method deletes an event based on a given Event ID and returns server's response indicating the operation status.

RockTF committed:
Merge remote-tracking branch 'origin/163-Delete-Event' into 163-Delete-Event

# Conflicts:
#	core/src/main/java/greencity/controller/EventController.java
#	service-api/src/main/java/greencity/constant/ErrorMessage.java
#	service-api/src/main/java/greencity/service/EventService.java
#	service/src/main/java/greencity/service/EventServiceImpl.java
